### DISCUSSION TITLE
Table cell width as defined by Polarin

### DISCUSSION THREAD

**Discussion by 77dhe** [SchweizerischeBundesbahnen/ch.sbb.polarion.extension.pdf-exporter#30] [Q&A] [answered]
We need for our documents more control over the table coloums width in the exported documents. At the moment the extension sets the width of each table cell to "auto" and left only the table cell hight at the value as defined by Polarion.
When I saw it right, this is processed in HtmlProcessor.java in adjustCellWidth(html).

What we need is, that the height of each table cell is set to "auto" and that the width value is left as defined by Polarion.

My proposal/request:

- Introduce an new Parameter "Keep table cell width" as boolean
- dependend on this parameter process the html in two ways:

1. As it is, when the parameter is not set
2. When the parameter is set: process the html and set the height of each table cell to "auto" instead of the width

Sorry, I am not able to make a code proposal for that.

**Comment by yurtsevich-sbb** [comment]
Hey,

First of all let me double check we are on the same line: you are talking about width of cells, **not height**, right? We are not explicitly manipulating height of cells, but only width. This was done by several reasons, but mainly for "Fit to page size" logic, though they are not explicitly linked with each other at the moment. Will it solve your request if we link them, i.e. column width manipulation will only happen if "Fit to page size" option is switched on?

**Reply by 77dhe** [reply]
Hey, thank you for your respose. Yes, I am talking about the width of the cells.

I think, your proposal will solve our request. But I think also, if the exporter don't process the cell width to "auto", it shall process the cell height" to "auto", so that the content of the cell can be dispayed without issues. 

As an example here is the html code from a table of the unprocessed html file:
`<td style="text-align: left;vertical-align: top;width: 269px;height: 58.5px;border: 1px solid #CCCCCC;padding: 5px;">`

procecced by the pdf-exporter with version 4.5:
`<td style="text-align: left;vertical-align: top;width: auto;height: 58.5px;border: 1px solid #CCCCCC;padding: 5px;">`

What we request:
`<td style="text-align: left;vertical-align: top;width: 269px;height: auto;border: 1px solid #CCCCCC;padding: 5px;">`

Make that sense for you?

**Reply by yurtsevich-sbb** [reply]
Hello,

OK, I got your idea from technical point of view, but to proceed with your proposal I would like to understand your use case. Can you describe your problem which you would like to solve?

**Reply by 77dhe** [reply]
We have the problem, that a auto-width for table columns does not always a good job in formating the column width. So we have to adjust the column width. But even if we do this in Polarion, it has no effect on the pdf, because the exporter removes our settings.

**Reply by yurtsevich-sbb** [reply]
Hi,
Thank you for the information. The problem is more or less clear now. But unfortunately I can't tell now how and when it will be implemented, as it's not yet defined in nearest road map.
(@ralfsmolarek3 referencing you FYI)

**Comment by grigoriev** [comment, marked as answer]
@77dhe,

we have introduced Webhooks for PDF-Exporter in v6.1.0.
Depending on your needs you can manipulate with HTML before sending it to WeasyPrint Service.

As example you can find the test implementation [here](https://github.com/SchweizerischeBundesbahnen/ch.sbb.polarion.extension.pdf-exporter.webhook-samples)

Probably this can satisfy your needs and solve your request.
